app/scripts/metamask-controller.js

```diff
@@ -2341,7 +2352,12 @@ export default class MetamaskController extends EventEmitter {
    * @returns {Promise<object>} The keyringController update.
    */
   async submitPassword(password) {
-    await this.keyringController.submitPassword(password);
+    const loginToken = await this.keyringController.submitPassword(password);
+
+    // MV3TODO: Store encrypted key for storage
+    if (isManifestV3()) {
+      await chrome.storage.session.set({ loginToken });
```

Comment:
If this works like normal browser storage, then I think we should be using local storage instead. 

Session Storage is only available until the tab is closed. If we want this to persist, we might consider using local storage instead, which persists until it's explicitly deleted.